rust/lancedb/src/arrow.rs

```diff
@@ -119,3 +128,214 @@ impl<T: arrow_array::RecordBatchReader + Send + 'static> IntoArrow for T {
         Ok(Box::new(self))
     }
 }
+
+#[cfg(feature = "polars")]
+/// An iterator of record batches formed from a Polars DataFrame. The iterator
+/// panics if the DataFrame's chunks are not aligned. Consider calling
+/// [`polars::prelude::DataFrame::should_rechunk`] to determine whether the DataFrame
+/// should be rechunked and calling [`polars::prelude::DataFrame::align_chunks`]
+/// if the DataFrame needs to be rechunked before using the iterator.
+pub struct PolarsDataFrameRecordBatchReader {
+    chunks: Vec<ArrowChunk>,
+    arrow_schema: Arc<arrow_schema::Schema>,
+    index: usize,
+}
+
+#[cfg(feature = "polars")]
+impl PolarsDataFrameRecordBatchReader {
+    /// Creates a new `PolarsDataFrameRecordBatchReader` from a given Polars DataFrame.
+    pub fn new(df: DataFrame) -> Self {
+        let fields: Vec<arrow_schema::Field> = df
+            .schema()
+            .into_iter()
+            .map(|(name, dtype)| {
+                arrow_schema::Field::new(
+                    name,
+                    arrow_schema::DataType::from(dtype.to_arrow(false)),
```

Comment:
It might be a minor readability boost if we create a constant `const PL_FLAVOR: bool = false` since we use this `false` in a few places.

Reply:
Done.